### ISSUE TITLE
Cannot Register Users - Invalid Session

### ISSUE THREAD

**Issue by AYJAYY** [calzoneman/sync#505]
Ubuntu 14.04
Node 0.12
Port 8080 using IP for domain currently.

I am getting "Invalid Session Your browser attempted to submit form data to /register with an invalid authentication token. This may be because:....." when I attempt to register a user. I have changed the cookie secret and cleared my cookies.

The server throws no errors and the error log is not showing anything. I searched through the issues and there doesn't seem to be anyone else with this issue.

**Comment by calzoneman** [comment]
This comes up quite a bit with people on IRC.  The most likely cause is that you have not set the cookie domain correctly in your config.yaml, hence the CSRF prevention cookie is unable to be set/read correctly.  Please check your configuration, and if you can't find the problem, then you can paste the configuration here (remove passwords first) and I'd be glad to take a look at it for you.

**Comment by AYJAYY** [comment]
Oh my, I cannot believe I missed that. I apologize for taking your time and I thank you for responding so quickly! That was my issue!

**Comment by AYJAYY** [comment]
~ Issue Solved - For others - make sure the "root-domain" in the cookies section of the config is set to your current domain. Thanks again!

**Comment by 70hundert** [comment]
Hi there!

I have the same issue, on "register" I get the "invalid authentication token"-message.

After I found this post here I followed the suggestion to set the "root-domain" in the config.yaml (same as,I set the "domain"), but the error is still there. I tried it in different browsers as well. 

Any further hint on this will be highly appreciated!

**Comment by calzoneman** [comment]
@70hundert can you sanitize your configuration (remove passwords) and link it so that I can take a look?

**Comment by 70hundert** [comment]
Hello Calvin!

Thanks for your quick reply!

Here you can see the config.yaml, which content I copied into this file:

http://slopetube.koding.io/config.txt

I only xxx'ed-out the mysql-password and the youtube-api.

**Comment by calzoneman** [comment]
Ah I see, the root domain is just supposed to be the domain itself, without any http:// prefix, since this is how the cookie header expects it.  I guess the comments in the config template don't make that clear.

**Comment by calzoneman** [comment]
So in your case, it should be `slopetube.koding.io`

**Comment by 70hundert** [comment]
Thanks again for the fast response!

So I edited my config.yaml file as you suggested, I changed line 53 to:

```
root-domain: 'slopetube.koding.io'
```

So now the registering and login is working! Great!

Then I was able to create a new channel for testing purposes. But by entering this channel, now I get the following error-message:

> Uh oh!
> It appears the connection to http://localhost:1337 has failed. If this error persists, a firewall or  antivirus is likely blocking the connection, or the server is down.

I also can't use the search in youtube-searchnor adding a video by url; also any Channel-setting I made didn't get saved, for example setting the channel being liste publicy. But I guess this is related to the error?
